src/coreclr/jit/emitarm64.h

```diff
@@ -658,8 +658,29 @@ static code_t insEncodeSplitUimm(size_t imm)
     return result;
 }
 
-// Returns the encoding for the immediate value as 4-bits at bit locations '19-16'.
-static code_t insEncodeSimm4_19_to_16(ssize_t imm);
+template <const size_t hi, const size_t lo>
+static code_t insEncodeSimm(ssize_t imm)
+{
+    // lo <= hi < 32
+    static_assert((hi >= lo) && (hi < sizeof(code_t) * BITS_PER_BYTE));
+
+    const size_t imm_bits = hi - lo + 1;
```

Comment:
Nit: Since you're statically asserting `imm_bits`'s value, should we make it `constexpr`?